app/models/miq_worker.rb

```diff
@@ -455,6 +455,17 @@ def clean_active_messages
     end
   end
 
+  def error_out_tasks_with_active_queue_message
+    message = "Task Handler: [#{friendly_name}] ID [#{id}] has been deleted!"
+    processed_messages.each do |m|
+      task = m.miq_task
```

Comment:
Can we preload all the tasks for all of the messages?  Otherwise, this is an N+1

Reply:
Actually from what I see you don't actually need the task object.  If you can get away with just the miq_task_id you don't need to queryy the second table at all.  Down below you can just put `:class_name => "MiqTask"`, because you don't actually need the specific subclass if you have the task id.

Reply:
yeah, good point.  If I queue put, I can just use miq_tasks_id and throw it on the queue.  If I do it directly, I can preload and instantiate the the task on each message.

Reply:
I started with doing it directly and moved to queuing it.  Either way, I'll have to go all the way one or the other and not mix them.  🤣